src/fsharp/vs/IncrementalBuild.fs

```diff
@@ -507,11 +507,11 @@ module internal IncrementalBuild =
     /// Visit each executable action necessary to evaluate the given output (with an optional slot in a
     /// vector output). Call actionFunc with the given accumulator.
     let ForeachAction (Target(output, optSlot)) bt (actionFunc:Action->'acc->'acc) (acc:'acc) =
-        let seen = Dictionary<Id,bool>()
+        let seen = System.Collections.Concurrent.ConcurrentDictionary<Id, unit>()
```

Comment:
wondering why it was taking so much CPU? is it worth to implement our concurrent hashset to not waste space for the value?